### DISCUSSION TITLE
Unique constraint - only one row can have flag set

### DISCUSSION THREAD

**Discussion by brielov** [prisma/prisma#4197] [Q&A] [answered]
I have a `companies` table and I wish to have a column which can be set to `true` on a single row at a time, like a single company can be flagged `true`. Can this be accomplished using prisma schema?

**Comment by ryands17** [comment, marked as answer]
@brielov 
I don't think that's possible with any constraint in SQL as far as I know. This would have to be done on the end of your application logic where you set the companies flag to `true` and the rest to false.

**Reply by brielov** [reply]
According to stackoverflow, in raw sql would be something like `create unique index on "public"."Company" (flag) where flag = true;`

**Reply by ryands17** [reply]
@brielov Ohh yeah, those constraints are possible, but currently you would need to set them via raw SQL itself. Prisma will respect any constraint errors caused and will throw an exception.

There is another way this will be possible via Prisma Migrate and SQL as there would be a stable version soon as we are getting very close to an initial stable release for Migrate!
If you want to get early access to the upcoming versions, you can fill out [this](https://prisma103696.typeform.com/to/yssOzU8F) Typeform and let us know.

**Comment by JasonWeise** [comment]
Sorry to resurect an old thread, but can this be done via Prisma schema now (original post 2020).
I have a table where I would like to only have one row in the table to be able have a "true" value via some sort of constraint on the field.

**Reply by tylermcrobert** [reply]
@JasonWeise hello from 2024. Were you able to accomplish this?

**Reply by JasonWeise** [reply]
Hello @tylermcrobert, unfortunately the future is not that bright. I moved on from Prisma and the JS ecosystem and threw in the towel.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
